starter-core/src/main/java/io/micronaut/starter/feature/build/maven/templates/gradleEnterprise.rocker.raw

```diff
@@ -1,19 +1,26 @@
-@args(String server)
-<gradleEnterprise>
+@import io.micronaut.starter.feature.build.gradle.GradleEnterpriseConfiguration
+@args (GradleEnterpriseConfiguration config)
+<gradleEnterprise xmlns="https://www.gradle.com/gradle-enterprise-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                          xsi:schemaLocation="https://www.gradle.com/gradle-enterprise-maven https://www.gradle.com/schema/gradle-enterprise-maven.xsd">
+@if (config.getServer() != null) {
     <server>
-        <url>@server</url>
+        <url>@(config.getServer())</url>
+        @if (config.allowTrustedServer() != null && config.allowTrustedServer()) {
+            <allowUntrusted>true</allowUntrusted>
+        }
     </server>
+}
     <buildScan>
+    @if (config.getServer() == null && (config.getTermsOfServiceUrl() != null || config.getTermsOfServiceAgree() != null)) {
+        <termsOfService>
+        @if (config.getTermsOfServiceUrl() != null) {
+            <url>@(config.getTermsOfServiceUrl())</url>
+        }
+        @if (config.getTermsOfServiceAgree() != null && config.getTermsOfServiceAgree()) {
+            <accept>true</accept>
+        }
+        </termsOfService>
+    }
         <publish>ALWAYS</publish>
     </buildScan>
-    <buildCache>
```

Comment:
I think this `buildCache` configuration was used via the starter build with the `MicronautGradleEnterprise` feature.  

I don't think we should remove them.

Reply:
I have added the buildCache back here: 
https://github.com/micronaut-projects/micronaut-starter/pull/1595/commits/7064a1daa2606d00eead912f81dc8ca0b2a2b8a5

@alextu are you ok with this change?

Reply:
So my intention was to adapt the snippet to the latest GE components: 

> When using Maven extension version 1.15+, Gradle Enterprise 2022.3+ and Gradle Enterprise Build Cache Nodes 13+, the same access key credential used for Build Scan publishing, Test Distribution and other functions can be used for build cache access control.

And since the build cache is enabled by default, there's no need to add a `buildCache` element at all. 
That being said I'm not familiar with how is used this `MicronautGradleEnterprise` feature, feel free to keep it if you think it's still required

Reply:
@alextu this is how we use it: 

https://github.com/micronaut-projects/micronaut-starter/blob/3.8.x/test-utils/src/main/groovy/io/micronaut/starter/test/CommandSpec.groovy#L124-L127

We add this hidden feature to the maven applications we generate as part of the functional test suite. The goal of this is to reduce the test execution time. 

thoughts @timyates @alvarosanchez ?

Reply:
To clarify a bit: with this change, given the build has the GE access key provisionned, building with the remote build cache should work as it currently does with the credentials specified. If it's not the case:
- passing the access key would be best (and checking the build cache read permissions are set for this key)
- if not possible, I would parametrized this template so setting the build cache credentials is the exception for this particular use case, not the norm

Does that make sense ?

Reply:
Exactly, note also that the provisionned Gradle enterprise access key associated user should also have "Read build cache data" and "Write build cache data" permissions. Best would be to check such a build on this branch, could you point me to one ?